frontend/src/components/EmployeeModal/Upload.tsx

```diff
@@ -1,34 +1,39 @@
 import React, { useState, createRef } from 'react';
 import uploadBox from './upload.svg';
 import styles from './employeemodal.module.css';
+//import { useToast, ToastStatus } from '../../context/toastContext';
+
+const IMAGE_SIZE_LIMIT = 50000;
 
 type UploadProps = {
   imageChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
   existingPhoto?: string;
 };
 
 const Upload = ({ imageChange, existingPhoto }: UploadProps) => {
+  //const { showToast } = useToast();
   const [imageURL, setImageURL] = useState(
     existingPhoto ? `${existingPhoto}` : ''
   );
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
   const inputRef = createRef<HTMLInputElement>();
+
   /* This is for accessibility purposes only */
   const handleKeyboardPress = (e: React.KeyboardEvent) => {
     if (e.key === 'Enter') {
       inputRef.current && inputRef.current.click();
     }
   };
+
   function previewImage(e: React.ChangeEvent<HTMLInputElement>) {
-    e.preventDefault();
     const { files } = e.target;
-    if (files && files[0] && files[0].size < 500000) {
-      const file = files[0];
-      const photoURL = URL.createObjectURL(file);
-      setImageURL(photoURL);
+    if (files && files[0] && files[0].size < IMAGE_SIZE_LIMIT) {
+      setImageURL(URL.createObjectURL(files[0]));
+      imageChange(e);
     } else {
-      console.log('invalid file');
+      setErrorMessage(`Images must be under ${IMAGE_SIZE_LIMIT / 1000} KB`);
```

Comment:
Hey Dwain, I checked out this branch and was having some issues uploading an image at first, so I checked the logs and got "Images must be under 50 KB," which seems to have come from here. After choosing a smaller image (20 KB) I was able to upload an image but the first one I tried was 116 KB, which is a size I think users should also be able to use to upload, so either here or in a future PR, maybe try either increasing the `IMAGE_SIZE_LIMIT` or investigating why the if statement would be returning false here.
Otherwise, good work on figuring out the backend and getting the uploading images to work!!

Reply:
Hello, the img size limit was set prior, in server/app.ts its set to 5000kb and anything above it throws a payload size error. I can ask Desmond about changing that directly. Otherwise I'll look into other ways to get past it when i add the optimizations.